### ISSUE TITLE
Atraxa's Skitterfang causes a null pointer exception

### ISSUE THREAD

**Issue by sprangg** [magefree/mage#10115]
After choosing a target for [[Atraxa's Skitterfang]]'s ability, the ongoing game suffers an unrecoverable null pointer exception.
```
java.io.NotSerializableException: java.util.HashMap$KeySet
        at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1184)
        at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
        at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
        at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
        at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
        at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
        at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
        at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
        at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
        at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
        at mage.remote.traffic.ZippedObjectImpl.zip(ZippedObjectImpl.java:31)
        at mage.remote.traffic.ZippedObjectImpl.<init>(ZippedObjectImpl.java:23)
        at mage.utils.CompressUtil.compress(CompressUtil.java:56)
        at mage.interfaces.callback.ClientCallback.setData(ClientCallback.java:57)
        at mage.interfaces.callback.ClientCallback.<init>(ClientCallback.java:26)
        at mage.interfaces.callback.ClientCallback.<init>(ClientCallback.java:20)
        at mage.server.game.GameSessionPlayer.lambda$chooseChoice$5(GameSessionPlayer.java:89)
        at java.util.Optional.ifPresent(Optional.java:159)
        at mage.server.game.GameSessionPlayer.chooseChoice(GameSessionPlayer.java:88)
        at mage.server.game.GameController.lambda$chooseChoice$17(GameController.java:825)
        at mage.server.game.GameController.perform(GameController.java:988)
        at mage.server.game.GameController.perform(GameController.java:981)
        at mage.server.game.GameController.chooseChoice(GameController.java:825)
        at mage.server.game.GameController.lambda$init$3aa0d15c$2(GameController.java:217)
        at mage.game.events.EventDispatcher.fireEvent(EventDispatcher.java:28)
        at mage.game.events.PlayerQueryEventSource.chooseChoice(PlayerQueryEventSource.java:92)
        at mage.game.GameImpl.fireChooseChoiceEvent(GameImpl.java:2798)
        at mage.player.human.HumanPlayer.choose(HumanPlayer.java:490)
        at mage.cards.a.AtraxasSkitterfangEffect.apply(AtraxasSkitterfang.java:107)
        at mage.abilities.AbilityImpl.resolveMode(AbilityImpl.java:200)
        at mage.abilities.AbilityImpl.resolve(AbilityImpl.java:184)
        at mage.abilities.TriggeredAbilityImpl.resolve(TriggeredAbilityImpl.java:151)
        at mage.game.stack.StackAbility.resolve(StackAbility.java:89)
        at mage.game.GameImpl.resolve(GameImpl.java:1635)
        at mage.game.GameImpl.playPriority(GameImpl.java:1561)
        at mage.game.turn.Step.priority(Step.java:61)
        at mage.game.turn.Phase.playStep(Phase.java:184)
        at mage.game.turn.Phase.play(Phase.java:89)
        at mage.game.turn.Turn.play(Turn.java:132)
        at mage.game.GameImpl.playTurn(GameImpl.java:1056)
        at mage.game.GameImpl.play(GameImpl.java:966)
        at mage.game.GameImpl.start(GameImpl.java:942)
        at mage.server.game.GameWorker.call(GameWorker.java:34)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:750)

**Comment by sprangg** [comment]
[Argivian Avenger](https://github.com/magefree/mage/blob/45bbfa10b76612b35d66e1f2959956a9fc7362ca/Mage.Sets/src/mage/cards/a/ArgivianAvenger.java) and [Steel Seraph](https://github.com/magefree/mage/blob/cf9a4a4eabc8bde454bc0e26d76e864247f61c24/Mage.Sets/src/mage/cards/s/SteelSeraph.java) also suffer from this NPE. I don't quite understand why but the problem happens in the ```player.choose``` part.

A similar, slightly differently implemented card that works is [Shifting Ceratops](https://github.com/magefree/mage/blob/4806626ba0ca78d4f582518a3df5c6b5ec070e0e/Mage.Sets/src/mage/cards/s/ShiftingCeratops.java).

**Comment by JayDi85** [comment]
It was `NotSerializableException` error related to `data.KeySet()` usage -- `HashMap` uses inner `KeySet` class without serialization support, so it's can't be used in client-server data (e.g. as choices). Must pass param as `new HashSet<>(data.keySet())`.

New checks and fixes added in 6b05562336158a6cd5d21c36e99a2c3b065b5c2c
